Error in React vite project due to rollup dependency (module not found)

whenever i ran docker compose up --build > then either i just add rollup manually to my package.json as devDependencies or by adding an optional dependency like [this](https://stackoverflow.com/questions/77569907/error-in-react-vite-project-due-to-rollup-dependency-module-not-found) 

 > [frontend frontend-builder 9/10] RUN npm run build: 0.464 0.464 > kubestellarui@0.0.0 build 0.464 > tsc -b && vite build 0.464 14.87 /app/node_modules/rollup/dist/native.js:64 14.87 throw new Error( 14.87 ^ 14.87 14.87 Error: Cannot find module @rollup/rollup-linux-arm64-gnu. npm has a bug related to optional dependencies (https://github.com/npm/cli/issues/4828). Please try npm i again after removing both package-lock.json and node_modules directory. 14.87 at requireWithFriendlyError (/app/node_modules/rollup/dist/native.js:64:9) 14.87 at Object.<anonymous> (/app/node_modules/rollup/dist/native.js:73:76) 14.87 ... 3 lines matching cause stack trace ... 14.87 at Module._load (node:internal/modules/cjs/loader:1096:12) 14.87 at cjsLoader (node:internal/modules/esm/translators:298:15) 14.87 at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:240:7) 14.87 at ModuleJob.run (node:internal/modules/esm/module_job:263:25) 14.87 at async ModuleLoader.import (node:internal/modules/esm/loader:540:24) { 14.87 [cause]: Error: Cannot find module '@rollup/rollup-linux-arm64-gnu' 14.87 Require stack: 14.87 - /app/node_modules/rollup/dist/native.js 14.87 at Module._resolveFilename (node:internal/modules/cjs/loader:1212:15) 14.87 at Module._load (node:internal/modules/cjs/loader:1043:27) 14.87 at Module.require (node:internal/modules/cjs/loader:1298:19) 14.87 at require (node:internal/modules/helpers:182:18) 14.87 at requireWithFriendlyError (/app/node_modules/rollup/dist/native.js:46:10) 14.87 at Object.<anonymous> (/app/node_modules/rollup/dist/native.js:73:76) 14.87 at Module._compile (node:internal/modules/cjs/loader:1529:14) 14.87 at Module._extensions..js (node:internal/modules/cjs/loader:1613:10) 14.87 at Module.load (node:internal/modules/cjs/loader:1275:32) 14.87 at Module._load (node:internal/modules/cjs/loader:1096:12) { 14.87 code: 'MODULE_NOT_FOUND', 14.87 requireStack: [ '/app/node_modules/rollup/dist/native.js' ] 14.87 } 14.87 } 14.87 14.87 Node.js v20.19.4 ------ failed to solve: process "/bin/sh -c npm run build" did not complete successfully: exit code: 1 * The terminal process "/bin/zsh '-l', '-c', 'docker compose -f 'docker-compose.yml' up -d --build 'frontend''" terminated with exit code: 1. * Terminal will be reused by tasks, press any key to close it.

--- a/frontend/Dockerfile
+++ b/frontend/Dockerfile
@@ -15,6 +15,7 @@ RUN git rev-parse HEAD | cut -c1-7 > commit_hash.txt 2>/dev/null || echo "unknow
 
 # Copy package files for caching
 COPY frontend/package.json frontend/package-lock.json ./
+RUN rm -rf node_modules package-lock.json || true
 RUN npm install --legacy-peer-deps
 
 # Accept build arguments

